### PR TITLE
feat: add missing geography field

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

Added missing "geography" attribute

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by rajatagarwal-ibm** [comment]
PR failed with the following error - added prefix to the string to avoid it in the future.

```
Error: CreateZoneWithContext failed. A network zone with the same name already exists. Assign each network zone a unique name.
{
    "StatusCode": 409,
    "Headers": {
        "Akamai-Grn": ["0.9dce4917.1725289640.10814556"],
        "Cache-Control": ["no-cache,no-store"],
        "Content-Length": ["317"],
        "Content-Type": ["application/json; charset=utf-8"],
        "Date": ["Mon, 02 Sep 2024 15:07:20 GMT"],
        "Expires": ["Thursday, 1 January 1970 00:00:00 GMT"],
        "Pragma": ["no-cache"],
        "Strict-Transport-Security": ["max-age=31536000; includeSubDomains"],
        "Transaction-Id": ["425b383c-60ce-4787-b208-4b9f8855f3b6"],
        "X-Content-Type-Options": ["nosniff"],
        "X-Correlation-Id": ["425b383c-60ce-4787-b208-4b9f8855f3b6"],
        "X-Proxy-Upstream-Service-Time": ["475"],
        "X-Ratelimit-Limit": ["10"],
        "X-Ratelimit-Remaining": ["9"],
        "X-Ratelimit-Reset": ["1725289641"]
    },
    "Result": {
        "errors": [
            {
                "code": "zone_already_exists_in_store",
                "message": "A network zone with the same name already exists. Assign each network zone a unique name.",
                "target": {
                    "name": "name",
                    "type": "field",
                    "value": "codeengine-zone-example-of-customized-zone-name"
                }
            }
        ],
        "status_code": 409,
        "trace": "425b383c-60ce-4787-b208-4b9f8855f3b6"
    },
    "RawResult": null
}

Error: CreateZoneWithContext failed. A network zone with the same name already exists. Assign each network zone a unique name.
{
    "StatusCode": 409,
    "Headers": {
        "Akamai-Grn": ["0.076bdc17.1725289640.3dcf8d31"],
        "Cache-Control": ["no-cache,no-store"],
        "Content-Length": ["310"],
        "Content-Type": ["application/json; charset=utf-8"],
        "Date": ["Mon, 02 Sep 2024 15:07:20 GMT"],
        "Expires": ["Thursday, 1 January 1970 00:00:00 GMT"],
        "Pragma": ["no-cache"],
        "Strict-Transport-Security": ["max-age=31536000; includeSubDomains"],
        "Transaction-Id": ["6b2f528e-36b0-4ddf-8805-359ae7d94d40"],
        "X-Content-Type-Options": ["nosniff"],
        "X-Correlation-Id": ["6b2f528e-36b0-4ddf-8805-359ae7d94d40"],
        "X-Proxy-Upstream-Service-Time": ["128"],
        "X-Ratelimit-Limit": ["10"],
        "X-Ratelimit-Remaining": ["0"],
        "X-Ratelimit-Reset": ["1725289641"]
    },
    "Result": {
        "errors": [
            {
                "code": "zone_already_exists_in_store",
                "message": "A network zone with the same name already exists. Assign each network zone a unique name.",
                "target": {
                    "name": "name",
                    "type": "field",
                    "value": "COS-zone-example-of-customized-zone-name"
                }
            }
        ],
        "status_code": 409,
        "trace": "6b2f528e-36b0-4ddf-8805-359ae7d94d40"
    },
    "RawResult": null
}

```

File: modules/fscloud/variables.tf
Comment (Ak-sky) [inline modules/fscloud/variables.tf:1]: I think we also need a validation for a service if it supports `geography` like we have a validation for `region`.

**Review by Ak-sky — CHANGES_REQUESTED** [review]

**Comment by rajatagarwal-ibm** [comment]
@Ak-sky I don't understand, custom names could have prefixes as well, right? I need to create that because it failed due to the naming conflict, see the error in the comment above.

**Comment by rajatagarwal-ibm** [comment]
@Ak-sky, I think we might need to remove the `region` attribute. I have added this topic to discuss with the wider audience.

File: modules/fscloud/variables.tf [hunk 1]
```diff
@@ -297,11 +299,16 @@ variable "target_service_details" {
...
+      for target_service_name, attributes in var.target_service_details : (
+        contains(["cloud-object-storage", "codeengine", "container-registry", "containers-kubernetes", "containers-kubernetes-cluster", "containers-kubernetes-management", "databases-for-cassandra", "databases-for-elasticsearch", "databases-for-enterprisedb", "databases-for-etcd", "databases-for-mongodb", "databases-for-mysql", "databases-for-postgresql", "databases-for-redis", "event-notifications", "hs-crypto", "iam-identity", "is", "logdna", "logdnaat", "messagehub", "messages-for-rabbitmq", "mqcloud", "secrets-manager", "sysdig-monitor", "sysdig-secure"], target_service_name) &&
+        (
+          !(attributes.region != null && attributes.geography != null)
+        )
+      )
     ])
-    error_message = "Provide a valid target service name that supports region attribute."
+    error_message = "Both `region` and `geography` cannot be set simultaneously for the target service."
```
Comment (Ak-sky) [inline modules/fscloud/variables.tf:309]: Geography is only applicable for `container-registry` and not for any other service. So validation has to be set specifically for `container-registry`.

```
Tgt. Services REGION SUPPORTED: 23
["cloud-object-storage", "codeengine", "container-registry", "containers-kubernetes", "databases-for-elasticsearch", "databases-for-enterprisedb", "databases-for-etcd", "databases-for-mongodb", "databases-for-mysql", "databases-for-postgresql", "databases-for-redis", "event-notifications", "hs-crypto", "iam-identity", "is", "logdna", "logdnaat", "messagehub", "messages-for-rabbitmq", "mqcloud", "secrets-manager", "sysdig-monitor", "sysdig-secure"]

Tgt. Services REGION NOT SUPPORTED: 13
["IAM", "apprapp", "compliance", "context-based-restrictions", "directlink", "dns-svcs", "globalcatalog-collection", "iam-access-management", "iam-groups", "kms", "schematics", "transit", "user-management"]

Tgt. Services GEOGRAPHY SUPPORTED: 1
["container-registry"]
```

File: examples/fscloud/main.tf [hunk 2]
```diff
@@ -100,19 +100,23 @@ module "cbr_account_level" {
       "enforcement_mode" = "report"
       "global_deny"      = false
     }
+    "container-registry" : {
```
Comment (vburckhardt) [inline examples/fscloud/main.tf:103]: I think we should just document this in the example README.md as well

**Review by Ak-sky — APPROVED** [review]

**Pull request merged**